news/_posts/2020-02-04-the-new-prestashop-test-framework.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: post
+title:  "The (new) PrestaShop Test Framework"
+subtitle: "And how we’re continually improving the quality of the PrestaShop project"
+date:  2020-02-05 12:00:00
+authors: [SimonGarny]
+icon: icon-checked
+image: /assets/images/theme/meta-logo-build.png
+tags: [QA, tests, community]
+---
+
+For a few months, quality has been a priority for PrestaShop: the QA team and I are proud to announce that we’re making great steps in improving the overall quality of the core project. Automatizing tests is a big part of this process: let’s see what we’re doing, why, and how!
+Note: this article is focused on the Core only, but that’s just a small part of what the QA team is working on. More to come in a future article!
+
+
+## What was there before
+
+Historically, the PrestaShop Core continuous integration had a big end-to-end test campaign which purpose was to test the whole application as thoroughly as possible. This campaign was using Selenium as the automation tool, and we were running it in two places:
+
+- The whole campaign was running on every nightly build and for every release
+- A small subset of the campaign was running for every Pull Request on GitHub through Travis CI
+
+At the beginning of 2019, the QA team decided to drop this campaign and start over. In September 2019, we started the new version.
```

Comment:
I dont understand what "we started the new version." means. The new version of what ? And when I read this, I wonder "so you dropped it in the beginning of 2019 and only started something new in September ? what happened in the 8 months in-between" ?

Reply:
At the beginning of 2019 we decided to stop improving the Selenium campaign (only maintenance) and create a new campaign using Puppeteer. It took quite some time to have a good POC and we started really working on it in September. Mid-September we had a sanity campaign up on Travis and in January we finally dropped the nightly Selenium campaign and replaced it with our growing Puppeteer one. 

I should clear that up, it's not precise enough.